core/flyout_base.js

```diff
@@ -612,10 +612,11 @@ Blockly.Flyout.prototype.createBlock_ = function(blockXml) {
  * @private
  */
 Blockly.Flyout.prototype.getBlockXml_ = function(blockInfo) {
-  var blockXml = null;
+  var blockXml = blockInfo['blockxml'];
```

Comment:
Can you cast this to a Node for clarity of type.

Reply:
At this point it could be either an element or a string, I can cast it when returning?

Reply:
Yeah, the desired type of blockXml is to be of type `Element` here so it's okay to cast at the top. You'd be saying, assume the type is Element.
You then check if the type is string below, and if so, convert it into an Element. 

The goal from a readability perspective, is to determine what the type of blockXml is, and that's not clear until further reading, so although the type cast isn't necessary, better to add if it helps readability

Reply:
This makes the compiler angry, since I am passing something that is an element to textToDom. I updated to try and make it more clear, but lmk if it doesn't and I can try again.